src/schema/slack/functions/_scripts/src/utils.ts

```diff
@@ -0,0 +1,115 @@
+import {
+  ParameterDefinition,
+} from "../../../../../parameters/definition_types.ts";
+import {
+  ManifestFunctionParameters,
+  ManifestFunctionSchema,
+} from "../../../../../manifest/manifest_schema.ts";
+import {
+  DefineFunctionInput,
+  FunctionParameter,
+  FunctionProperty,
+  FunctionRecord,
+  FunctionsPayload,
+} from "./types.ts";
+
+const FUNCTIONS_JSON_PATH = "functions.json";
+const DIRECTORY_PATH_ALLOW_LIST = ["_scripts", "mod.ts"];
+
+const green = "\x1b[92m";
+const yellow = "\x1b[38;5;214m";
+const red = "\x1b[91m";
+const reset = "\x1b[0m";
+
+export const greenText = (text: string) => green + text + reset;
+export const yellowText = (text: string) => yellow + text + reset;
+export const redText = (text: string) => red + text + reset;
+
+export function isValidFunctionFile(fileName: string) {
+  return !(fileName.endsWith("_test.ts") ||
+    DIRECTORY_PATH_ALLOW_LIST.includes(fileName));
+}
+
+export async function loadFunctionsJson(): Promise<FunctionsPayload> {
+  return await Deno.readTextFile(
+    FUNCTIONS_JSON_PATH,
+  ).then(JSON.parse);
+}
+
+export function getDefineFunctionInputs(
+  functionsPayload: FunctionsPayload,
+): DefineFunctionInput[] {
+  // Filter out any non slack functions (i.e. has an app_id) parse to DefineFunctionInput
+  return functionsPayload.functions.reduce<DefineFunctionInput[]>(
+    (result, fn) => {
+      if (!fn.app_id) {
+        result.push(getDefineFunctionInput(fn));
+      }
+      return result;
+    },
+    [],
+  );
+}
+
+const getParameterDefinition = (
+  param: FunctionProperty,
+): ParameterDefinition => {
+  // deno-lint-ignore no-explicit-any
+  const paramDef: { [key: string]: any } = {
```

Comment:
Not required, but FYI there is also the typescript utility type [`Record<KeyType, ValueType>`](https://www.typescriptlang.org/docs/handbook/utility-types.html#recordkeys-type) that you can use instead of `{ [key: string}: any }`, e.g. `Record<string, any>`.